ui/src/plugins/dev.perfetto.MetricsPage/metrics_page.ts

```diff
@@ -48,7 +53,40 @@ async function getMetric(
   }
 }
 
-class MetricsController {
+async function getMetricV2(
+  engine: Engine,
+  metric: string,
+  metricId: string,
+  _format: Format,
+): Promise<string> {
+  const result = await engine.summarizeTrace(
+    [metric],
+    [metricId],
+    undefined,
+    _format === 'proto' ? 'proto' : 'prototext',
+  );
+  if (result.error || result.error.length > 0) {
+    throw new Error(result.error);
+  }
+  switch (_format) {
+    case 'json':
+      if (!result.protoSummary) {
+        throw new Error('Error fetching Textproto trace summary');
+      }
+      return JSON.stringify(result.protoSummary, null, 2);
+    case 'prototext':
+      if (!result.textprotoSummary) {
+        throw new Error('Error fetching Textproto trace summary');
+      }
+      return result.textprotoSummary;
+    case 'proto':
+      throw new Error('Proto format not supported');
+    default:
+      throw new Error(`Unknown format ${_format}`);
```

Comment:
You can use `assertUnreachable(_format)` here which will assert at build time that you've covered all possible values.